docstrings_to_openapi/httpdomain.py

```diff
@@ -0,0 +1,265 @@
+from dataclasses import dataclass
+import sys
+
+from .utils import (
+    prepare_docstring,
+    remove_start_and_end_empty_strings
+)
+
+
+FORM_KINDS       = ['formparameter', 'formparam', 'fparam', 'form']
+JSON_KINDS       = ['jsonparameter', 'jsonparam', 'json']
+PARAM_KINDS      = ['param', 'parameter', 'arg', 'argument']
+QUERY_KINDS      = ['queryparameter', 'queryparam', 'qparam', 'query']
+REQJSON_KINDS    = ['reqjsonobj', 'reqjson', '<jsonobj', '<json']
+RESJSON_KINDS    = ['resjsonobj', 'resjson', '>jsonobj', '>json']
+REQJSONARR_KINDS = ['reqjsonarr', '<jsonarr']
+RESJSONARR_KINDS = ['resjsonarr', '>jsonarr']
+REQHEADER_KINDS  = ['requestheader', 'reqheader', '>header']
+RESHEADER_KINDS  = ['responseheader', 'resheader', '<header']
+STATUS_KINDS     = ['statuscode', 'status', 'code']
+
+
+@dataclass
+class Directive:
+    kind: str
+    type: str
+    name: str
+    body: str
+    properties: list
+
+
+def _split_list_by_function(l, func):
+    """For each item in l, if func(l) is truthy, func(l) will be added to l1.
+    Otherwise, l will be added to l2.
+    """
+    l1 = []
+    l2 = []
+    for item in l:
+        res = func(item)
+        if res:
+            l1.append(res)
+        else:
+            l2.append(item)
+    return l1, l2
+
+
+def _parse_directive(directive_line):
+    first_colon = directive_line.find(':')
+    second_colon = directive_line.find(':', first_colon + 1)
+
+    if first_colon != 0 or second_colon is None:
+        return None
+
+    # If we don't have a type, just default it to string
+    _type = 'string'
```

Comment:
Same underscore thing here, why is `_type` the only one with an underscore?

Reply:
well for this one, cause `type` is a builtin function, so not to mangle the builtin with the thing we use for this, not sure of a better name